gradoop-flink/src/test/java/org/gradoop/flink/model/impl/operators/matching/single/cypher/operators/join/JoinEmbeddingsTest.java

```diff
@@ -0,0 +1,647 @@
+package org.gradoop.flink.model.impl.operators.matching.single.cypher.operators.join;
+
+import com.google.common.collect.Lists;
+import org.apache.flink.api.java.DataSet;
+import org.gradoop.common.model.impl.id.GradoopId;
+import org.gradoop.common.model.impl.properties.Properties;
+import org.gradoop.flink.model.impl.operators.matching.single.cypher.common.pojos.Embedding;
+import org.gradoop.flink.model.impl.operators.matching.single.cypher.common.pojos.EmbeddingEntry;
+import org.gradoop.flink.model.impl.operators.matching.single.cypher.common.pojos.IdEntry;
+import org.gradoop.flink.model.impl.operators.matching.single.cypher.common.pojos.ProjectionEntry;
+import org.gradoop.flink.model.impl.operators.matching.single.cypher.operators.PhysicalOperator;
+import org.gradoop.flink.model.impl.operators.matching.single.cypher.operators.PhysicalOperatorTest;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@SuppressWarnings("ArraysAsListWithZeroOrOneArgument")
+public class JoinEmbeddingsTest extends PhysicalOperatorTest {
+
+  //------------------------------------------------------------------------------------------------
+  // Test embedding extensions
+  //------------------------------------------------------------------------------------------------
+
+  /**
+   * Tests the extension of a vertex embedding with an edge embedding.
+   *
+   * [Id(v0)]
+   * |><|(0=0)
+   * [Id(v0),Id(e0),Id(v1)]
+   * ->
+   * [Id(v0),Id(e0),Id(v1)]
+   */
+  @Test
+  public void testEdgeJoinOnRightmostColumn() throws Exception {
+    GradoopId v0 = GradoopId.get();
+    GradoopId v1 = GradoopId.get();
+
+    GradoopId e0 = GradoopId.get();
+
+    // [Id(v0)]
+    ArrayList<EmbeddingEntry> leftEntries = Lists.newArrayList(new IdEntry(v0));
```

Comment:
Should we have a `DataSet<Embedding> createEmbeddings(int size, EmbeddingEntry... entries)` as well